### PR TITLE
[RDY] document correct init.vim path for Windows

### LINKED ISSUE #7374: Documentation specifies incorrect `init.vim` path on Windows
The documentation on [transitioning to nvim from vim](https://neovim.io/doc/user/nvim.html#nvim-from-vim) states that `~/.vimrc` should be symlinked or copied to `~/.config/nvim/init.vim`, when in fact (on my machine at least) NeoVim apepars only to look for `C:\Users\<username>\AppData\Local\nvim\init.vim` (which is typically `~/AppData/Local/nvim/init.vim`).

If the path changes depending on configuration or version, an easy way to check where NeoVim is looking for config files is `:echo &rtp`.

### PR DESCRIPTION
**Pull request by euclio**
Fixes #7374.

### CODE REVIEW

**Comment by justinmk** [comment]
Also: we can do this, but it will be temporary. There are 2 steps needed to reduce the initial friction:

1. `:write foo/bar/baz.txt` should auto-create the `foo/bar/` directorys if they don't exist.
2. `$MYVIMRC` should always just have the damn filepath in it. 

After those steps are implemented, new users can just do this:

    :edit $MYVIMRC
    :write

This simplifies documentation and eliminates paragraphs of explanation.

**Comment by euclio** [comment]
Fixed. I'm fine with this being temporary.

**Comment by jamessan** [comment]
> `$MYVIMRC` should always just have the damn filepath in it.

That sounds reasonable.

> `:write foo/bar/baz.txt` should auto-create the `foo/bar/` directorys if they don't exist.

This I'm not so sure about.  I think it should at least ask the user, rather than doing it automatically.

As far as this PR, though, 👍 for the doc clarification @euclio.

File: runtime/doc/nvim.txt [hunk 1]
```diff
@@ -20,15 +20,24 @@ differences from Vim.
...
 Transitioning from Vim				*nvim-from-vim*
 
-To start the transition, create ~/.config/nvim/init.vim with these contents:
+To start the transition, create init.vim in the correct directory for your
+platform.
+
+For Linux, macOS and other Unixes, create the file at ~/.config/nvim/init.vim.
+
+For Windows, create the file at %LOCALAPPDATA%\nvim\init.vim.`%LOCALAPPDATA%`
+usually expands to `C:\Users\<username>\AppData\Local\nvim\init.vim`.
```
[the remark this answers is not in the document]
Comment (euclio) [reply runtime/doc/nvim.txt]: Good catch. Fixed.

**Pull request merged**